diff --git a/check-stateful b/check-stateful.sh.in
index 218103c..35b856a 100644
--- a/check-stateful
+++ b/check-stateful.sh.in
@@ -3,18 +3,19 @@
 # Usage: check-stateful SRCDIR CHARSET
 srcdir="$1"
 charset="$2"
+test_exe=@TEST_EXE@
 set -e
 
 # charset, modified for use in filenames.
 charsetf=`echo "$charset" | sed -e 's,:,-,g'`
 
 if test -f "${srcdir}"/"$charsetf"-snippet.alt; then
-  ../src/iconv_no_i18n -f "$charset" -t UTF-8 < "${srcdir}"/"$charsetf"-snippet.alt > tmp-snippet
+  $test_exe -f "$charset" -t UTF-8 < "${srcdir}"/"$charsetf"-snippet.alt > tmp-snippet
   cmp "${srcdir}"/"$charsetf"-snippet.UTF-8 tmp-snippet
 fi
-../src/iconv_no_i18n -f "$charset" -t UTF-8 < "${srcdir}"/"$charsetf"-snippet > tmp-snippet
+$test_exe -f "$charset" -t UTF-8 < "${srcdir}"/"$charsetf"-snippet > tmp-snippet
 cmp "${srcdir}"/"$charsetf"-snippet.UTF-8 tmp-snippet
-../src/iconv_no_i18n -f UTF-8 -t "$charset" < "${srcdir}"/"$charsetf"-snippet.UTF-8 > tmp-snippet
+$test_exe -f UTF-8 -t "$charset" < "${srcdir}"/"$charsetf"-snippet.UTF-8 > tmp-snippet
 cmp "${srcdir}"/"$charsetf"-snippet tmp-snippet
 rm -f tmp-snippet
 exit 0
diff --git a/check-stateless b/check-stateless.sh.in
index 7104ea1..355698c 100644
--- a/check-stateless
+++ b/check-stateless.sh.in
@@ -5,14 +5,17 @@ srcdir="$1"
 charset="$2"
 set -e
 
+TABLE_FROM=@TABLE_FROM@
+TABLE_TO=@TABLE_TO@
+
 # charset, modified for use in filenames.
 charsetf=`echo "$charset" | sed -e 's,:,-,g'`
 
 # iconv in one direction.
-./table-from "$charset" > tmp-"$charsetf".TXT
+$TABLE_FROM "$charset" > tmp-"$charsetf".TXT
 
 # iconv in the other direction.
-./table-to "$charset" | sort > tmp-"$charsetf".INVERSE.TXT
+$TABLE_TO "$charset" | sort > tmp-"$charsetf".INVERSE.TXT
 
 # Check 1: charmap and iconv forward should be identical.
 cmp "${srcdir}"/"$charsetf".TXT tmp-"$charsetf".TXT 2> /dev/null
diff --git a/check-translit b/check-translit.sh.in
index babc91e..ce9cd0b 100644
--- a/check-translit
+++ b/check-translit.sh.in
@@ -5,8 +5,9 @@ srcdir="$1"
 file="$2"
 fromcode="$3"
 tocode="$4"
+test_exe=@TEST_EXE@
 set -e
-../src/iconv_no_i18n -f "$fromcode" -t "$tocode"//TRANSLIT < "${srcdir}"/"$file"."$fromcode" > tmp
+$test_exe -f "$fromcode" -t "$tocode"//TRANSLIT < "${srcdir}"/"$file"."$fromcode" > tmp
 cmp "${srcdir}"/"$file"."$tocode" tmp
 rm -f tmp
 exit 0
diff --git a/check-subst b/check-subst.sh.in
index 1b5f387..cadb65f 100644
--- a/check-subst
+++ b/check-subst.sh.in
@@ -1,7 +1,7 @@
 #!/bin/sh
 # Check of --unicode-subst, --byte-subst, --widechar-subst options.
 set -e
-iconv=../src/iconv_no_i18n
+iconv=@TEST_EXE@
 
 options_ascii='--unicode-subst=<U+%04X> --byte-subst=<0x%02x> --widechar-subst=<%08x>'
 options_utf8='--unicode-subst=«U+%04X» --byte-subst=«0x%02x» --widechar-subst=«%08x»'
diff --git a/check-translitfailure b/check-translitfailure.sh.in
index 37dd86e..1c166cf 100644
--- a/check-translitfailure
+++ b/check-translitfailure.sh.in
@@ -5,6 +5,7 @@ srcdir="$1"
 file="$2"
 fromcode="$3"
 tocode="$4"
-../src/iconv_no_i18n -f "$fromcode" -t "$tocode"//TRANSLIT < "${srcdir}"/"$file"."$fromcode" >/dev/null 2>/dev/null
+test_exe=@TEST_EXE@
+$test_exe -f "$fromcode" -t "$tocode"//TRANSLIT < "${srcdir}"/"$file"."$fromcode" >/dev/null 2>/dev/null
 test $? = 1
 exit $?
